djangocms_link/forms.py

```diff
@@ -6,6 +6,14 @@
 from .fields import PageSearchField
 from .models import Link
 
+from .compat import CMS_LT_4
+
+if CMS_LT_4:
```

Comment:
This logic doesn't do anything here, this should be inside the function: `for_site` currently line 29

Reply:
Like that?

```python
from django.forms.models import ModelForm
from django.utils.translation import gettext_lazy as _

from djangocms_attributes_field.widgets import AttributesWidget

from .fields import PageSearchField
from .models import Link

class LinkForm(ModelForm):
    internal_link = PageSearchField(
        label=_('Internal link'),
        required=False,
    )

    def for_site(self, site):
        # override the internal_link fields queryset to contains just pages for
        # current site
        # this will work for PageSelectFormField
        from cms.models import Page
        self.fields['internal_link'].queryset = Page.objects.on_site(site)
        # set the current site as a internal_link field instance attribute
        # this will be used by the field later to properly set up the queryset
        # this will work for PageSearchField
        self.fields['internal_link'].site = site
        self.fields['internal_link'].widget.site = site

        from .compat import CMS_LT_4

        if CMS_LT_4:
            # django CMS 3 compatibility
            self.fields['internal_link'].queryset = Page.objects.drafts().on_site(site)
            else
            # django CMS 4.0.x + compatibility
            self.fields['internal_link'].queryset = Page.objects.on_site(site)


    class Meta:
        model = Link
        exclude = ('page', 'position', 'placeholder', 'language', 'plugin_type')

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fields['attributes'].widget = AttributesWidget()
```

Reply:
@crydotsnake The import `from .compat import CMS_LT_4` should be with the other imports at the top of the file.

also the `else` for the if statement: `if CMS_LT_4:` has lost its formatting and appears to be indented too far.

Reply:
Sorry!. I really need too learn python 😅 

This should be okay:

```python
from django.forms.models import ModelForm
from django.utils.translation import gettext_lazy as _

from djangocms_attributes_field.widgets import AttributesWidget

from .fields import PageSearchField
from .models import Link
from .compat import CMS_LT_4

class LinkForm(ModelForm):
    internal_link = PageSearchField(
        label=_('Internal link'),
        required=False,
    )

    def for_site(self, site):
        # override the internal_link fields queryset to contains just pages for
        # current site
        # this will work for PageSelectFormField
        from cms.models import Page
        self.fields['internal_link'].queryset = Page.objects.on_site(site)
        # set the current site as a internal_link field instance attribute
        # this will be used by the field later to properly set up the queryset
        # this will work for PageSearchField
        self.fields['internal_link'].site = site
        self.fields['internal_link'].widget.site = site

        if CMS_LT_4:
            # django CMS 3 compatibility
            self.fields['internal_link'].queryset = Page.objects.drafts().on_site(site)
        else:
            # django CMS 4.0.x + compatibility
            self.fields['internal_link'].queryset = Page.objects.on_site(site)

    class Meta:
        model = Link
        exclude = ('page', 'position', 'placeholder', 'language', 'plugin_type')

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.fields['attributes'].widget = AttributesWidget()
```

Reply:
I pushed the changes